module/zfs/spa_misc.c

```diff
@@ -1809,6 +1816,12 @@ spa_get_slop_space(spa_t *spa)
 	 * the entire pool.
 	 */
 	slop = MAX(slop, MIN(space >> 1, spa_min_slop));
+
+	/*
+	 * Additionally, slop space should never exceed spa_max_slop.
+	 */
+	slop = MIN(slop, spa_max_slop);
```

Comment:
To be extra precise, on medium to large pools, I think that the space that is unavailable today is the sum of:
* the 4MB labels/boot region per vdev (also not included in the zpool SIZE property)
* the metaslab rounding (up to 16GB on large vdevs) per vdev (also not included in the zpool SIZE property)
* 1/32nd of the remainder (which is the sum of the embedded log metaslab and the return value of spa_get_slop_space()) (included in zpool SIZE but not zfs USED+AVAIL)

Considering the interaction with the embedded log class, I think that the proposed change is that for large pools (vdev size >2TB, total pool size >4TB), the space that's unavailable will be the sum of:
* the 4MB labels/boot region per vdev (also not included in the zpool SIZE property)
* the metaslab rounding (up to 16GB per vdev) (also not included in the zpool SIZE property)
* the embedded log metaslab (16GB per vdev) (included in zpool SIZE but not zfs USED+AVAIL)
* 128GB (included in zpool SIZE but not zfs USED+AVAIL)

The embedded log metaslab change kept the **user-visible slop space (the difference between zpool SIZE and zfs USED+AVAIL)** the same.  The proposed change exposes this, so the user-visible slop space will depend on number of vdevs and the value of `zfs_embedded_slog_min_ms`.

We might want to keep the user-visible slop space constant regardless of the embedded log (on most configurations).  For example we could set the default spa_max_slop to 256GB, but have spa_get_slop_space() return MAX(128GB, 256GB - embedded_log_space).  That way with up to 8 top-level vdevs, the spa_max_slop exactly equals the user-visible slop space.  To implement this, we would just move the new code before the `embedded_log` adjustment.